queue/queue.go

```diff
@@ -126,31 +103,40 @@ func (qm *Manager) DeclareQueue() error {
 	if err != nil {
 		return err
 	}
-	if qm.Logger != nil {
+	if qm.logger != nil {
 		qm.LogInfo("queue declared")
 	}
-	qm.Queue = &q
+	qm.queue = &q
 	return nil
 }
 
-// ConsumeMessage is used to consume messages that are sent to the queue
+// ConsumeMessages is used to consume messages that are sent to the queue
 // Question, do we really want to ack messages that fail to be processed?
 // Perhaps the error was temporary, and we allow it to be retried?
-func (qm *Manager) ConsumeMessage(consumer, dbPass, dbURL, dbUser string, cfg *config.TemporalConfig) error {
+func (qm *Manager) ConsumeMessages(ctx context.Context, wg *sync.WaitGroup, consumer string, sslModeDisable bool, cfg *config.TemporalConfig) error {
 	db, err := database.OpenDBConnection(database.DBOptions{
```

Comment:
shouldnt this db and stuff be part of the main constructor, or are there instances of the queue where you dont need the DB?

Reply:
if the latter, you might want to pass around the `db` instead of making the various message consumption helpers hope that `qm` is initialized with a database manager, which makes the dependency more explicit. For example, init the DB here, and then:

```go
	case DatabaseFileAddQueue:
		return qm.ProcessDatabaseFileAdds(ctx, wg, db, msgs)
```

even better, since this will probably be initialized alongside a bunch of other stuff, perhaps `ConsumeMessages` can accept a database as a parameter as well? So as to create the connection once, and pass it off to classes that depend on it

Reply:
imo it's good practice to streamline dependency trickle-downs, and make sure we know who needs what, with the idea that each class should need as little as possible